Dart/thirdPartySrc/analysisServer/org/dartlang/analysis/server/protocol/AnalysisError.java

```diff
@@ -119,12 +119,12 @@ public boolean equals(Object obj) {
   public static AnalysisError fromJson(JsonObject jsonObject) {
     String severity = jsonObject.get("severity").getAsString();
     String type = jsonObject.get("type").getAsString();
-    String code = jsonObject.get("code") == null ? null : jsonObject.get("code").getAsString();
     Location location = Location.fromJson(jsonObject.get("location").getAsJsonObject());
     String message = jsonObject.get("message").getAsString();
     String correction = jsonObject.get("correction") == null ? null : jsonObject.get("correction").getAsString();
+    String code = jsonObject.get("code").getAsString();
```

Comment:
It's just this particular field which is the issue. Given the current generated Java impl., adding a new, non-optional field to an existing class is a breaking change. I don't know if you speak to what's considered a breaking change in the protocol spec doc, but it would strike me as strange to support missing required fields.

I think marking the field as optional is a pretty low-cost solution. I'm happy to add some docs to the field - `only optional for backwards compatibility reasons`. What do you think?